modules/apps/portal-security-sso/portal-security-sso-openid-connect-impl/src/main/java/com/liferay/portal/security/sso/openid/connect/internal/OpenIdConnectAuthenticationHandlerImpl.java

```diff
@@ -137,8 +138,37 @@ public void processAuthenticationResponse(
 			_getLoginRedirectURI(httpServletRequest),
 			oAuthClientEntry.getTokenRequestParametersJSON());
 
-		String userInfoJSON = _requestUserInfoJSON(
-			oidcTokens.getAccessToken(), oidcProviderMetadata);
+		URI userInfoURI = oidcProviderMetadata.getUserInfoEndpointURI();
+
+		String userInfoJSON = null;
+
+		if (userInfoURI == null) {
+				JWT idToken = oidcTokens.getIDToken();
+
+				JWTClaimsSet userInfoClaimSet = idToken.getJWTClaimsSet();
+
+				List<String> emails = userInfoClaimSet.getStringListClaim(
+					"emails");
+				String email = emails.get(0);
```

Comment:
You need a blank line before this line. Running `gradlew formatSource` should report it?

Reply:
sourceFormatter does not work for me right now :-(